File: VB/App_Code/DataHelper.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Public NotInheritable Class DataHelper

    Private Sub New()
    End Sub

    Private Shared ReadOnly Property _data() As List(Of Model)
        Get
            If HttpContext.Current.Session("data") Is Nothing Then
                HttpContext.Current.Session("data") = GenerateData()
            End If

            Return DirectCast(HttpContext.Current.Session("data"), List(Of Model))
        End Get
    End Property
    Public Shared ReadOnly Property Data() As List(Of Model)
        Get
            Return _data
        End Get
    End Property
    Private Shared Function GenerateData() As List(Of Model)
        Dim list = Enumerable.Range(0, 10).Select(Function(x) New Model With { _
            .Oid = x, _
            .IsRead = If(x <> 3 AndAlso x <> 5, False, True), _
            .Title = "Title" & x, _
            .Message = "Message" & x _
        }).ToList()
        Return list
    End Function
    Public Shared Sub MakeAllUnread()
        For Each item In Data
            item.IsRead = False
        Next item
    End Sub
    Public Shared Sub MarkAsRead(ByVal ID As Integer)
        Data.Find(Function(x) x.Oid = ID).IsRead = True
    End Sub
End Class

File: CS/App_Code/DataHelper.cs
using System.Collections.Generic;
using System.Linq;
using System.Web;

public static class DataHelper {
	static List<Model> _data { get {
			if(HttpContext.Current.Session["data"] == null)
				HttpContext.Current.Session["data"] = GenerateData();

			return (List<Model>)HttpContext.Current.Session["data"];
		}
	}
	public static List<Model> Data {
		get { return _data; }		
	}
	private static List<Model> GenerateData() {
		var list = Enumerable.Range(0, 10).Select(x => new Model {
			Oid = x,
			IsRead = (x != 3 && x != 5) ? false : true,
			Title = "Title" + x,
			Message = "Message" + x
		}).ToList();
		return list;
	}
	public static void MakeAllUnread() {		
		foreach(var item in Data) {
			item.IsRead = false;
		}		
	}
	public static void MarkAsRead(int ID) {		
		Data.Find(x => x.Oid == ID).IsRead = true;		
	}
}